src/main/groovy/com/palantir/gradle/shadowjar/ShadowJarPlugin.java

```diff
@@ -111,6 +112,13 @@ private void setupShadowJarToShadeTheCorrectDependencies(
             conf.setCanBeResolved(false);
         });
 
+        // Tell the consistent versions plugin not to look at the unshaded configuration for the
+        // checkUnusedConstraints task,  It forces resolution before the configuration below is complete which causes
+        // an error under gradle8.
```

Comment:
I think this plugin requires GCV to work properly - will need to think whether this actually keeps it working?

Reply:
We lock unshaded below... not sure what that means without version props injection. Perhaps it works out fine as we still get the constraints from elsewhere as it locked?

Reply:
My reasoning is this - The unshaded configuration is setup automatically based on the inputs of other configurations.  Therefore the dependencies will be resolved as part of those other configurations.  Similarly, the unshaded configuration is not going to have unused constraints that are not already resolved within those other configurations.

Reply:
I think that's enough to convince me, just need to use `excludeConfigurationFromVersionsPropsInjection` and we can merge this in